cmd/prompt.go

```diff
@@ -0,0 +1,167 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	prompt "github.com/c-bata/go-prompt"
+	"github.com/kris-nova/kubicorn/cutil/logger"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var commandFlagSuggestions = make(map[string][]prompt.Suggest)
+var commandMap = make(map[string][]prompt.Suggest)
+var promptSuggestions = []prompt.Suggest{}
+var promptCmd = &cobra.Command{
+	Use:   "prompt",
+	Short: "Open a prompt with auto-completion (non-Windows)",
+	Long: `Use this command to use the Kubicron API via a shell prompt.
+
+This command will open a prompt using go-prompt (with auto-completion) to
+allow you to run commands interactively from the shell.
+Currently this doesn't work on Windows systems`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) > 0 {
+			logger.Critical("Too many arguments.")
+			os.Exit(1)
+		}
+		err := RunPrompt()
+		if err != nil {
+			logger.Critical(err.Error())
+			os.Exit(1)
+		}
+	},
+}
+
+func RunPrompt() error {
+	fmt.Println("Please input your kubicorn commands.")
+	defer fmt.Println("Bye!")
+
+	p := prompt.New(
+		executor,
+		completer,
+		prompt.OptionTitle("kubicorn: interactive kubicorn client"),
+		prompt.OptionPrefix(">> "),
+		prompt.OptionInputTextColor(prompt.Yellow))
+
+	p.Run()
+	return nil
+}
+
+func executor(s string) {
+	s = strings.TrimSpace(s)
+	if s == "" {
+		return
+	} else if s == "quit" || s == "exit" {
+		fmt.Println("Bye!")
+		os.Exit(0)
+		return
+	}
+
+	cmd := exec.Command("/bin/sh", "-c", "kubicorn "+s)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		fmt.Printf("Got error: %s\n", err.Error())
+	}
+	return
+}
+
+func completer(d prompt.Document) []prompt.Suggest {
+	if d.TextBeforeCursor() == "" {
+		return []prompt.Suggest{}
+	}
+	args := strings.Split(d.TextBeforeCursor(), " ")
+	w := d.GetWordBeforeCursor()
+
+	// If PIPE is in text before the cursor, returns empty suggestions.
+	for i := range args {
+		if args[i] == "|" {
+			return []prompt.Suggest{}
+		}
+	}
+
+	// If word before the cursor starts with "-", returns CLI flag options.
+	if strings.HasPrefix(w, "-") {
+		return optionCompleter(args)
+	}
+
+	return argumentsCompleter(excludeOptions(args))
+}
+
+func argumentsCompleter(args []string) []prompt.Suggest {
+	if len(args) <= 1 {
+		return prompt.FilterHasPrefix(promptSuggestions, args[0], true)
+	}
+	first := args[0]
+	return prompt.FilterHasPrefix(promptSuggestions, first, true)
+}
+
+func excludeOptions(args []string) []string {
+	ret := make([]string, 0, len(args))
+	for i := range args {
+		if !strings.HasPrefix(args[i], "-") {
+			ret = append(ret, args[i])
+		}
+	}
+	return ret
+}
+
+func initializePrompt() {
```

Comment:
This approach has a problem, but let's leave it as it as, because it would require total refactor of `cmd` code base.

The problem is that you will not have completions for any command after `prompt` as it's still not initialized.

As I said, let's keep this as it is and open up the new issue once this PR is merged.

Reply:
Sorry, I don't completely understand the issue here. The initializePrompt is called in the init() function. This causes the prompt.go to initialize whenever kubicorn is invoked via `kubicorn prompt`. 
And when you say "refactor of cmd code base", does that mean refactoring only prompt.go or the way the entire package is organized.